lib/core/routing/routes_manager.dart

```diff
@@ -20,7 +24,22 @@ class RouteGenerator {
             child: const HomeScreen(),
           ),
         );
-
+      case Routes.rocketsRoute:
+        final rocketsResponse = settings.arguments as RocketsResponse?;
+        if (rocketsResponse == null) {
+          return unDefinedRoute();
+        }
+        final rocketModel = RocketModel.fromJson(rocketsResponse.toJson());
```

Comment:
why this line here in route ?

  final rocketModel = RocketModel.fromJson(rocketsResponse.toJson());

Reply:
A = in feature/home, it includes a "RocketsResponse"
B =  in feature/rocket, it includes a "RocketModel"
when navigating from the home screen to the Rockets screen
Send (A) as an argument but the Rockets screen needs to (B) so I used this line in the route,
because don't I use mode in (feature) to another (feature) to maintain feature independence

Who is responsible for turning A to B better than the route?

I'm thinking about something like this
```dart
extension RocketsResponseMapper on RocketsResponse {
  RocketModel toDomain() {
    return RocketModel.fromJson(this.toJson());
  }
}
```

Reply:
Yes!

you can use any mapper or mapping approach.